packages/select/src/Select/Select.tsx

```diff
@@ -43,563 +52,576 @@ import { DropdownWidthBasis, SelectProps, Size, State } from './Select.types';
-    [ariaLabelledby, ariaLabel, label, id],
-  );
-
-  if (!label && !ariaLabelledby && !ariaLabel) {
-    console.error(
-      'For screen-reader accessibility, label, aria-label, or aria-labelledby must be provided to Select.',
+export const Select = forwardRef<HTMLDivElement, SelectProps>(
+  (
+    {
+      children,
+      darkMode: darkModeProp,
+      size = Size.Default,
+      disabled = false,
+      allowDeselect = true,
+      usePortal = true,
+      placeholder = 'Select',
+      errorMessage = '',
+      state = State.None,
+      dropdownWidthBasis = DropdownWidthBasis.Trigger,
+      baseFontSize = BaseFontSize.Body1,
+      id: idProp,
+      'aria-labelledby': ariaLabelledby,
+      'aria-label': ariaLabel,
+      className,
+      label,
+      description,
+      name,
+      defaultValue,
+      value,
+      onChange,
+      readOnly,
+      portalContainer,
+      scrollContainer,
+      portalClassName,
+      popoverZIndex,
+      __INTERNAL__menuButtonSlot__,
+      ...rest
+    },
+    fwdRef,
+  ) => {
+    const id = useIdAllocator({ prefix: 'select', id: idProp });
+    const labelId = useMemo(
+      () => (ariaLabel && !label ? undefined : ariaLabelledby ?? `${id}-label`),
+      [ariaLabelledby, ariaLabel, label, id],
     );
-  }
-
-  const { darkMode } = useDarkMode(darkModeProp);
 
-  const descriptionId = `${id}-description`;
-  const menuId = `${id}-menu`;
-
-  const [open, setOpen] = useState(false);
+    if (!label && !ariaLabelledby && !ariaLabel) {
+      console.error(
+        'For screen-reader accessibility, label, aria-label, or aria-labelledby must be provided to Select.',
+      );
+    }
 
-  const menuButtonRef = useStateRef<HTMLDivElement | null>(null);
-  const menuButtonId = useIdAllocator({ prefix: 'select' });
-  const listMenuRef = useStateRef<HTMLUListElement | null>(null);
+    const { darkMode } = useDarkMode(darkModeProp);
 
-  const sizeSet = sizeSets[size];
+    const descriptionId = `${id}-description`;
+    const menuId = `${id}-menu`;
 
-  const providerData = useMemo(() => {
-    return { size, open, disabled };
-  }, [size, open, disabled]);
+    const [open, setOpen] = useState(false);
 
-  useEffect(() => {
-    if (value !== undefined && onChange === undefined && readOnly !== true) {
-      console.warn(
-        'You provided a `value` prop to a form field without an `onChange` handler. ' +
-          'This will render a read-only field. ' +
-          'If the field should be mutable use `defaultValue`. ' +
-          'Otherwise, set either `onChange` or `readOnly`.',
-      );
-    }
-  }, [onChange, readOnly, value]);
-
-  /**
-   * Open / close state
-   */
-  const onOpen = useCallback(() => {
-    setOpen(true);
-  }, []);
-
-  const onClose = useCallback(() => {
-    setOpen(false);
-    menuButtonRef.current!.focus();
-  }, [menuButtonRef]);
-
-  useEffect(() => {
-    if (!open) {
-      return;
-    }
+    const containerRef = useForwardedRef(fwdRef, null);
```

Comment:
is this the right place to forward teh ref?

Reply:
technically yes, should be on the container. I'd _prefer_ it to be on the button element itself, but then there's no way to  reference the Label/Description